test/test_util_network_ssl.cpp

```diff
@@ -676,6 +677,142 @@ TEST(Util_Network_SSL_BasicSendAndReceive)
 }
 
 
+#if REALM_HAVE_SECURE_TRANSPORT
+TEST(Util_Network_SSL_Nonzero_Length_Error)
+{
+    network::Service service_1;
+    network::Service service_2;
+    network::Socket socket_1{service_1};
+    network::Socket socket_2{service_2};
+    network::ssl::Context ssl_context_1;
+    network::ssl::Context ssl_context_2;
+    configure_server_ssl_context_for_test(ssl_context_1);
+    network::ssl::Stream ssl_stream_1 = {socket_1, ssl_context_1, network::ssl::Stream::server};
+    network::ssl::Stream ssl_stream_2 = {socket_2, ssl_context_2, network::ssl::Stream::client};
+    ssl_stream_1.set_logger(test_context.logger.get());
+    ssl_stream_2.set_logger(test_context.logger.get());
+    connect_ssl_streams(ssl_stream_1, ssl_stream_2);
+    network::ReadAheadBuffer rab;
+
+    const char* message = "hello";
+    char buffer[50];
+
+    bool shutdown_completed = false;
+    bool read_completed = false;
+    auto&& [r_promise, read_future] = util::make_promise_future<void>();
+    auto read_handler = [this, &read_completed, read_promise = std::move(r_promise)](std::error_code ec,
+                                                                                     std::size_t) mutable {
+        CHECK_EQUAL(util::MiscExtErrors::premature_end_of_input, ec);
+        read_promise.emplace_value();
+        read_completed = true;
```

Comment:
if you wanted, you could just use the fact that the read_promise has been filled by calling `read_future.is_ready()` instead of having an explicit tracking boolean.